pre_processing/extract_landscape.py

```diff
@@ -0,0 +1,143 @@
+                         [RIGHT_BROW, LEFT_BROW, NOSE, RIGHT_EYE, LEFT_EYE, MOUTH]
+        only_test_data -- returns facial vectors for only training data
+        only_test_data -- returns facial vectors for only test data
+
+    :return numpy array [n, 62, 2], where n is number of photos analysed
+    """
+    logger.debug("Getting facial vectors")
+    photos_matrix = _extract_photos_from_file(False, only_train_data, only_test_data)
+
+    facial_featutes_matrix = np.zeros([35886, 68, 2])
+    detector = dlib.get_frontal_face_detector()
+    predictor = dlib.shape_predictor(PREDICTOR_PATH)
+    faces_found = 0
+    faces_not_found = 0
+    logger.debug("Starting facial detection and feature extraction")
+    for i in range(35886):
+        img = photos_matrix[i, :, :]
+        dets = detector(img, 2)
+        if len(dets) != 0:
+            facial_featutes_matrix[i, :, :] = _get_vectors_of_image_from_image(dets, predictor, img)
+            faces_found += 1
+        else:
+            facial_featutes_matrix[i, :, :] = np.zeros([68, 2])
+            faces_not_found += 1
+        if i % 10000 == 0:
+            logger.debug("photos gone through {}".format(i))
+
+    print("faces found: {}   faces not found : {}".format(faces_found, faces_not_found))
+
+    return facial_featutes_matrix
+
+
+def _extract_photos_from_file(extract_first_only=False, only_train_data=False, only_test_data=False):
+    logger.debug("Extracting photos from csv")
+    if only_train_data:
+        images_matrix = np.zeros([TRAIN_END_POINT, 48, 48], dtype="uint8")
+    elif only_test_data:
+        images_matrix = np.zeros([PUBLIC_TEST_END_POINT - 28708, 48, 48], dtype="uint8")
+    else:
+        images_matrix = np.zeros([35886, 48, 48], dtype="uint8")
+
+    with open(DATA_CSV_FILE, newline='') as f:
+        reader = csv.reader(f)
+        _ = next(reader)
+        for k, pixels in enumerate(reader):
+            k_mod = k
+            image_csv_format = np.zeros([1, IMAGE_SIZE])
+            pixels_formated = [int(a) for a in pixels[1].split(" ")]
+            image_csv_format[0, :] = pixels_formated
+            image_csv_format = np.reshape(image_csv_format, [48, 48])
+            if k % 5000 == 0:
+                logger.debug("photos extracted from csv {}".format(k))
+            if only_train_data and k > TRAIN_END_POINT:
+                break
+            elif only_test_data & k < PUBLIC_TEST_START_POINT:
+                next
+                k_mod = k - PUBLIC_TEST_START_POINT
+            images_matrix[k_mod, :, :] = image_csv_format
+            if extract_first_only:
+                break
+
+    logger.debug("Finished extracting photos from csv")
+    return images_matrix
+
+
+def _initilise_gui_window(img):
+    win = dlib.image_window()
+    win.clear_overlay()
+    win.set_image(img)
+    return win
+
+
+def _get_vectors_of_image_from_image(dets, predictor, img, win=None):
+    for k, d in enumerate(dets):
```

Comment:
k is number of faces found in picture, would like to keep it incase we expand to include video streams as examples.